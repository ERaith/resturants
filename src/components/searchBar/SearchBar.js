import React, { useState } from "react";

export function SearchBar({ handleSearchChange }) {
  const [searchValue, setSearchValue] = useState("");
  const submit = (e) => {
    e.preventDefault();
    handleSearchChange(searchValue);
  };
  const clear = () => {
    setSearchValue("");
    handleSearchChange("");
  };
  return (
    <form onSubmit={submit}>
      <label htmlFor="search"> SEARCH </label>
      <input
        className="search"
        id="search"
        onChange={(e) => setSearchValue(e.target.value)}
        placeholder="Search"
        type="text"
        value={searchValue}
      />
      <button type="submit" value="Submit" className="submit-search">
        Submit
      </button>
      <button type="reset" className="clear-search" onClick={clear}>
        Clear Search
      </button>
    </form>
  );
}
