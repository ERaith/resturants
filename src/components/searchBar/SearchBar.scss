.submit-search {
  @include button-bg(#2ecc71);
}

.clear-search {
  @include button-bg(#e74c3c);
}

.search-container {
  background: #f9f9f9;
  border: 1px solid #c1c1c1;
  display: flex;
  margin-left: auto;
  margin-right: auto;
  margin-top: 2rem;
  padding: 1em;
  padding: 1em;

  input {
    background: #fff;
    border: 1px solid #9c9c9c;
    margin-bottom: 1rem;
    .search {
      width: 100%;
    }
    .checkbox {
      height: 2em;
      width: 2em;
    }
  }
  label {
    display: block;
    padding: 0.5em 1.5em 0.5em 0;
  }
  
  .filter-container {
    margin-left: auto;
    width: 30%;
    .filter {
      min-width: 8em;
    }
  }
}
