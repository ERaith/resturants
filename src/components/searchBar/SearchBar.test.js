import React from "react";
import { SearchBar } from "./SearchBar";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";

describe("SearchBar", () => {
  describe("Snapshots", () => {
    it("should match the snapshot", () => {
      const mockHandleSearchChange = jest.fn();
      const props = {
        handleSearchChange: mockHandleSearchChange,
      };
      const wrapper = mount(<SearchBar {...props} />);

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe("user interactions", () => {
    it("should update searchValue", () => {
      const mockHandleSearchChange = jest.fn();
      const props = {
        handleSearchChange: mockHandleSearchChange,
      };
      const wrapper = mount(<SearchBar {...props} />);
      const input = wrapper.find('input[type="text"]');

      expect(input.props()).toEqual({
        className: "search",
        id: "search",
        onChange: expect.anything(),
        placeholder: "Search",
        type: "text",
        value: "",
      });

      input.instance().value = "TEST";
      input.simulate("change");

      expect(input.instance().value).toBe("TEST");
      expect(wrapper.find("TEST")).toBeTruthy();
      expect(toJson(wrapper)).toMatchSnapshot();
    });
    it("should clear value when button is clicked", () => {
      const mockHandleSearchChange = jest.fn();
      const props = {
        handleSearchChange: mockHandleSearchChange,
      };
      const wrapper = mount(<SearchBar {...props} />);
      const input = wrapper.find('input[type="text"]');

      input.instance().value = "TEST";
      input.simulate("change");

      expect(input.instance().value).toBe("TEST");
      expect(wrapper.find("TEST")).toBeTruthy();

      const clear = wrapper.find('button[type="reset"]');
      clear.simulate("click");

      expect(input.instance().value).toBe("");
    });
    it("should submit value when button is clicked", () => {
      const mockHandleSearchChange = jest.fn();
      const props = {
        handleSearchChange: mockHandleSearchChange,
      };
      const wrapper = mount(<SearchBar {...props} />);
      const input = wrapper.find('input[type="text"]');

      input.instance().value = "TEST";
      input.simulate("change");

      expect(input.instance().value).toBe("TEST");
      expect(wrapper.find("TEST")).toBeTruthy();

      const submit = wrapper.find('button[type="submit"]');
      submit.simulate("submit");

      expect(input.instance().value).toBe("TEST");
      expect(mockHandleSearchChange).toHaveBeenCalledWith("TEST");
    });
  });
});
