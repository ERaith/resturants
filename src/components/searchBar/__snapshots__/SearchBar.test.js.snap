// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Snapshots should match the snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <label
    htmlFor="search"
  >
     SEARCH 
  </label>
  <input
    className="search"
    id="search"
    onChange={[Function]}
    placeholder="Search"
    type="text"
    value=""
  />
  <button
    className="submit-search"
    type="submit"
    value="Submit"
  >
    Submit
  </button>
  <button
    className="clear-search"
    onClick={[Function]}
    type="reset"
  >
    Clear Search
  </button>
</form>
`;

exports[`SearchBar user interactions should update searchValue 1`] = `
<SearchBar>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="search"
    >
       SEARCH 
    </label>
    <input
      className="search"
      id="search"
      onChange={[Function]}
      placeholder="Search"
      type="text"
      value="TEST"
    />
    <button
      className="submit-search"
      type="submit"
      value="Submit"
    >
      Submit
    </button>
    <button
      className="clear-search"
      onClick={[Function]}
      type="reset"
    >
      Clear Search
    </button>
  </form>
</SearchBar>
`;
